File: src/device.cpp
// src/device.cpp
#include "minml/device.h"

namespace minml {

namespace {
Device g_default = Device::CPU;
}

Device default_device() { return g_default; }
void set_default_device(Device d) { g_default = d; }

const char* device_name(Device d) {
  switch (d) {
    case Device::CPU: return "cpu";
    case Device::CUDA: return "cuda";
    case Device::WebGPU: return "webgpu";
  }
  return "unknown";
}

}  // namespace minml

File: src/device.ts
export enum Device {
  Cpu = "CPU",
  Cuda = "CUDA",
  WebGpu = "WebGPU",
}

let DEFAULT_DEVICE: Device = Device.Cpu;

export function default_device(): Device {
  return DEFAULT_DEVICE;
}

export function set_default_device(d: Device): void {
  DEFAULT_DEVICE = d;
}
